lib/modules/apostrophe-browser-utils/public/js/lean.js

```diff
@@ -110,7 +120,19 @@
           data = this.responseText;
         }
       }
-      return callback(null, data);
+      return callback(error(), data);
+      function error() {
+        // xmlhttprequest does not consider a 404, 500, etc. to be
+        // an "error" in the sense that would trigger the error
+        // event handler function (below). It looks like only network
+        // failures and/or browser problems can do that. So we need
+        // to recognize error status codes ourselves and correctly
+        // report an error.
+        if (xmlhttp.status < 400) {
+          return null;
+        }
+        return xmlhttp;
```

Comment:
A simple way to get an object with a status property (the xmlhttprequest object is already such an object).